include/sensors/core/tasks/capacitive_sensor_callbacks.hpp

```diff
@@ -98,18 +89,48 @@ struct ReadCapacitanceCallback {
         this->number_of_reads = number_of_reads;
     }
 
-    auto get_offset() -> float { return current_offset; }
+    [[nodiscard]] auto get_offset() const -> float { return current_offset_pf; }
 
     void set_echoing(bool should_echo) { echoing = should_echo; }
 
-    void set_bind_sync(bool should_bind) { bind_sync = should_bind; }
+    void set_bind_sync(bool should_bind) {
+        bind_sync = should_bind;
+        hardware.reset_sync();
+    }
+
+    void set_offset(float new_offset) {
+        if (new_offset != current_offset_pf) {
+            auto capdac_raw = get_capdac_raw(new_offset);
+            std::array offset{CONFIGURATION_MEASUREMENT,
+                              device_configuration_msb(capdac_raw),
+                              device_configuration_lsb(capdac_raw)};
+            i2c_writer.write(ADDRESS, offset);
+            current_offset_pf = new_offset;
+            std::array configuration_data{FDC_CONFIGURATION, SAMPLE_RATE_MSB,
+                                          SAMPLE_RATE_LSB};
+            i2c_writer.write(ADDRESS, configuration_data);
+        }
+    }
+
+    void initialize() {
+        current_offset_pf = -1;
```

Comment:
Did you just pick -1 or is there a reason to use this value specifically?

Is this the same as if you define `float current_offset_pf = 1;` in line 132 and then call `set_offset(0)`  here?

Reply:
that would work the same, yeah